Zenventory/Services/CoreDataService.swift

```diff
@@ -9,41 +9,39 @@ import Foundation
 import Combine
 import CoreData
 
-class CoreDataService: ObservableObject, CoreDataManager {
-    let container: NSPersistentContainer
-    @Published var savedEntities: [ProductEntity] = []
-    var savedEntitiesPublisher: Published<[ProductEntity]>.Publisher { $savedEntities }
-    var savedEntitiesPublished: Published<[ProductEntity]> { _savedEntities }
+final internal class CoreDataService: ObservableObject, CoreDataManager {
+    @Published internal var savedEntities: [ProductEntity] = []
+
+    internal let container: NSPersistentContainer
+    internal var savedEntitiesPublisher: Published<[ProductEntity]>.Publisher { $savedEntities }
+    internal var savedEntitiesPublished: Published<[ProductEntity]> { _savedEntities }
 
     private var cancellables = Set<AnyCancellable>()
 
-    init() {
+    internal init() {
         container = NSPersistentContainer(name: "ProductsContainer")
-        container.loadPersistentStores { (description, error) in
-            if let error = error {
-                print("Error loading core data: \(error.localizedDescription)")
-            }
-        }
-        fetchProducts()
+        container.loadPersistentStores { (_, _) in }
+        try? fetchProducts()
     }
 
-    func fetchProducts() {
+    internal func fetchProducts() throws {
         let request = NSFetchRequest<ProductEntity>(entityName: "ProductEntity")
 
         do {
             savedEntities = try container.viewContext.fetch(request)
         } catch {
-            print(error.localizedDescription)
+            throw error
         }
     }
 
-    func addProduct(name: String,
-                    guarantee: Int?,
-                    careName: String?,
-                    careInterval: Int?,
-                    price: Double?,
-                    importance: String) {
-
+    internal func addProduct(
+        name: String,
+        guarantee: Int?,
+        careName: String?,
+        careInterval: Int?,
+        price: Double?,
+        importance: String
+    ) {
         let newProduct = ProductEntity(context: container.viewContext)
```

Comment:
I've mentioned in the previous code review, there is IMO better way to work with types. I mean I'd provide types  explicitly for example: 
`let newProduct: ProductEntity = .init(context: container.viewContext)` 

But it's up to you :)

Reply:
However, if you decide to change it, review the entire application.